crates/rethnet_evm/benches/state_ref_basic.rs

```diff
@@ -0,0 +1,94 @@
+use criterion::{black_box, criterion_group, criterion_main, BatchSize, BenchmarkId, Criterion};
+use revm::primitives::AccountInfo;
+use std::str::FromStr;
+
+use rethnet_eth::Address;
+use rethnet_evm::state::{LayeredState, RethnetLayer};
+
+fn make_state(number_of_addresses: u64) -> LayeredState<RethnetLayer> {
+    let mut state = LayeredState::<RethnetLayer>::default();
+    for i in 1..number_of_addresses {
+        use rethnet_evm::state::StateDebug;
+        state
+            .insert_account(
+                Address::from_str(&format!("0x{:0>40x}", i)).unwrap(),
+                AccountInfo::default(),
+            )
+            .unwrap();
+        state.checkpoint().unwrap();
+    }
+    state
+}
+
+const ADDRESS_SCALES: [u64; 4] = [100, 500, 1000, 2000];
+
+fn bench_layered_create(c: &mut Criterion) {
+    for number_of_addresses in ADDRESS_SCALES.iter() {
```

Comment:
Could you set it up such that there is an additional for loop over the different types of `StateRef` implementations? That way we only need one function per benchmark and can reuse it for all (state type, no. addresses) pairs.

You can achieve this by having a closure that returns a `Box<dyn StateRef>`: `Vec<Box<dyn Fn(u64) -> Box<dyn StateRef>>>`

For now, we'll just have the `LayeredState`